Add Super Wireless Kit

lc-1337 opened the pull request:
TODO: update states.png before merge 

Have 3 modes, simple / advanced same as tools in ae2stuff. Super mode have gui, display wireless connectors/hubs from bound networks, allow bind, unbind and recolor single/grouped wireless connectors/hubs. 
TAB + Rclick for switch mode.

![image](https://github.com/user-attachments/assets/d6676db5-7029-40f8-968c-39679af31ed0)

![image](https://github.com/user-attachments/assets/50f60996-e9c0-4c00-82aa-99f8558b4e16)

![image](https://github.com/user-attachments/assets/37dfc48d-0f33-40b2-9f0f-d93293062367)

![image](https://github.com/user-attachments/assets/a9259e92-e023-4102-9c82-76ee999b1b61)

lc-1337 committed:
1st commit

lc-1337 committed:
add info for blocks

lc-1337 committed:
add hide bounded setting & button

lc-1337 committed:
add grower

lc-1337 committed:
move from main pr

lc-1337 committed:
DESUPERIZATION

lc-1337 committed:
backup wireless code

lc-1337 committed:
copilot review

lc-1337 committed:
textures

lc-1337 committed:
update states

lc-1337 committed:
Merge branch 'master' into ae2stuff

lc-1337 committed:
spotless

serenibyss committed:
Merge branch 'master' into ae2stuff

lc-1337 committed:
Merge branch 'ae2stuff' into super-wireless-kit

# Conflicts:
#	src/main/java/appeng/api/config/Settings.java
#	src/main/java/appeng/client/gui/widgets/GuiImgButton.java
#	src/main/java/appeng/core/api/definitions/ApiItems.java
#	src/main/java/appeng/core/localization/ButtonToolTips.java
#	src/main/java/appeng/core/sync/AppEngPacketHandlerBase.java
#	src/main/java/appeng/core/sync/GuiBridge.java
#	src/main/resources/assets/appliedenergistics2/lang/en_US.lang
#	src/main/resources/assets/appliedenergistics2/textures/guis/states.png

lc-1337 committed:
merge ae2stuff

serenibyss committed:
Merge branch 'master' into ae2stuff

serenibyss committed:
Merge branch 'master' into ae2stuff

serenibyss committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
review

lc-1337 committed:
spotless

lc-1337 committed:
Merge remote-tracking branch 'origin/ae2stuff' into super-wireless-kit

# Conflicts:
#	src/main/java/appeng/api/config/ActionItems.java
#	src/main/java/appeng/core/sync/GuiBridge.java

lc-1337 committed:
update

lc-1337 committed:
remove relict

lc-1337 committed:
Merge branch 'ae2stuff' into super-wireless-kit

lc-1337 committed:
spotless

lc-1337 committed:
fix off icons

lc-1337 committed:
Merge branch 'ae2stuff' into super-wireless-kit

lc-1337 committed:
cached item check

lc-1337 committed:
icon get

Dream-Master committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
Add button for convert connectors

lc-1337 committed:
Merge remote-tracking branch 'origin/super-wireless-kit' into super-wireless-kit

lc-1337 committed:
fix npe

lc-1337 committed:
Merge branch 'ae2stuff' into super-wireless-kit

lc-1337 committed:
fix lang

lc-1337 committed:
Merge branch 'master' into ae2stuff

lc-1337 committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
fix fuzzy stacking

lc-1337 committed:
Merge branch 'ae2stuff' into super-wireless-kit

lc-1337 committed:
fixes

NeOzay committed:
refactor implementation wireless connector from AE2Stuff: rename and restructure classes for clarity (WIP)

Dream-Master committed:
Merge branch 'master' into super-wireless-kit

# Conflicts:
#	dependencies.gradle
#	src/main/java/appeng/api/config/Settings.java
#	src/main/java/appeng/client/gui/widgets/GuiImgButton.java
#	src/main/java/appeng/core/localization/ButtonToolTips.java
#	src/main/java/appeng/core/sync/AppEngPacketHandlerBase.java
#	src/main/resources/assets/appliedenergistics2/lang/en_US.lang
#	src/main/resources/assets/appliedenergistics2/textures/guis/states.png

Dream-Master committed:
Merge branch 'master' into super-wireless-kit

NeOzay committed:
Merge branch 'fork/lc-1337/super-wireless-kit' into super-wireless-kit-ozay

Ethryan committed:
Try to get it to compile

by removing the symbol the compiler complains about.

NeOzay committed:
Merge branch 'master' into super-wireless-kit

NeOzay committed:
Refactor wireless connectivity logic: consolidate connection handling, improve clarity, and adapt naming conventions

NeOzay committed:
update WAILA dependency to implementation scope in dependencies.gradle

NeOzay committed:
share custom name between wireless

NeOzay committed:
spotless

NeOzay committed:
Refactor localization system: unify chat message handling with `Localization` interface and consolidate `toChat` methods across enums

NeOzay committed:
Refactor wireless tools and localization: unify message handling, simplify wireless mode management, and improve naming consistency

NeOzay committed:
finalize PR

NeOzay committed:
fix missing

NeOzay commented:
also require: https://github.com/GTNewHorizons/Applied-Energistics-2-Unofficial/pull/791

NeOzay committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
Merge remote-tracking branch 'origin/master' into fork/lc-1337/super-wireless-kit

# Conflicts:
#	dependencies.gradle
#	src/main/java/appeng/api/config/AdvancedWirelessToolMode.java
#	src/main/java/appeng/api/config/Settings.java
#	src/main/java/appeng/block/networking/BlockWirelessConnector.java
#	src/main/java/appeng/client/ClientHelper.java
#	src/main/java/appeng/client/gui/widgets/GuiImgButton.java
#	src/main/java/appeng/core/Registration.java
#	src/main/java/appeng/core/localization/ButtonToolTips.java
#	src/main/java/appeng/core/sync/AppEngPacketHandlerBase.java
#	src/main/java/appeng/core/sync/GuiBridge.java
#	src/main/java/appeng/crafting/MECraftingInventory.java
#	src/main/java/appeng/integration/modules/waila/TileWailaDataProvider.java
#	src/main/java/appeng/util/Platform.java
#	src/main/resources/assets/appliedenergistics2/lang/en_US.lang
#	src/main/resources/assets/appliedenergistics2/textures/guis/states.png

lc-1337 committed:
refactor

lc-1337 committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
fix

lc-1337 committed:
Merge remote-tracking branch 'origin/super-wireless-kit' into super-wireless-kit

lc-1337 committed:
fix

Ranzuu commented on src/main/java/appeng/client/gui/widgets/GuiColorButton.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    @Override
+    public void drawButton(Minecraft mc, int mouseX, int mouseY) {
```
Let's not do this please ;v

A texture would be way better in every way
Disabled/Normal/Hover + 9 slice

<img width="18" height="54" alt="Image" src="https://github.com/user-attachments/assets/89b66953-b649-4959-818a-f428ae048225" />

lc-1337 replied:
whats wrong with button?

Ranzuu replied:
It's fully hardcoded, aka no support for resource packs

lc-1337 replied:
button represent AEColor enum, which configurable, no?

Ranzuu replied:
AEColor is like only dye like color which is not configurable. So it's not. Either way also see the main message on my review

Ranzuu replied:
Update because I am on pc.
It's using `AEcolor`, while it should use `GuiColors`

lc-1337 replied:
But its a representation of colors that can be applied to connector. why they should can be different from ae colors?

Ranzuu replied:
I got some things confused, let me get this straighten up.
- `indicatorColor` to Enum, already good, thats nice thank you
- New button with hardcoded colors (GuiAeButton)
https://github.com/lc-1337/applied-energistics-2-unofficial/blob/26f4f9ab5bc564d4431ebeec3843b135d4886d03/src/main/java/appeng/client/gui/widgets/GuiAeButton.java#L34-L51
This should be changed to normal button we already have without hardcoded colors
- About the colors boxes like RED, GREEN, LIME etc. It is good

lc-1337 replied:
fixed

Ranzuu commented on src/main/java/appeng/client/gui/widgets/GuiColorButton.java:
```diff
@@ -0,0 +1,51 @@
+                    this.xPosition,
+                    this.xPosition + this.width - 1,
+                    this.yPosition + this.height - 1,
+                    0xFF404040);
```
You lied, no Color Enum 🎇 
0xFF404040

Ranzuu commented on src/main/java/appeng/client/gui/implementations/GuiSuperWirelessKit.java:
```diff
@@ -0,0 +1,1358 @@
+                    GuiColors.CraftingCPUStored.getColor());
+            GL11.glPopMatrix();
+
+            int indicatorColor = AEColor.Lime.mediumVariant;
```
please new Enum Color

Ranzuu commented on src/main/java/appeng/client/gui/implementations/GuiSuperWirelessKit.java:
```diff
@@ -0,0 +1,1358 @@
+                    GuiColors.CraftingCPUStored.getColor());
+            GL11.glPopMatrix();
+
+            int indicatorColor = AEColor.Lime.mediumVariant;
```
please new Enum Color

Ranzuu reviewed (commented):
Other side notes:
- confirm renaming does not work on NumPad "Enter" (also renamed slots only show on the 2nd and 3rd column)
- Please add all "indicatorColor" enums, don't use default AEColor variants :<
(Added 2 comments on just lime, but there is also Orange, Red or the negative overlay)


Now I'm thinking about the button, we have one already lol
`GuiAeButton`, just need to add hover state and not draw some `0xFF5C669D` lines

Kogepan229 commented on src/main/java/appeng/helpers/WireLessToolHelper.java:
```diff
@@ -0,0 +1,278 @@
+            throw new NullPointerException("dc is null"); // maybe return null instead
+        }
+
+        TileEntity te = w.getTileEntity(dc.x, dc.y, dc.z);
```
need to check dimension

lc-1337 replied:
check happens in performConnection, do you think its aslo need there?

Kogepan229 replied:
Ah I missed that.
However, checking it here makes the code clearer and helps prevent mistakes, so we should do it.

lc-1337 replied:
I rewrite bind function, check this out

Kogepan229 commented on src/main/java/appeng/tile/networking/TileWirelessConnector.java:
```diff
@@ -0,0 +1,101 @@
+
+    @Override
+    protected void tryRestoreConnection(List<DimensionalCoord> locList) {
+        if (connection == null) {
```
```suggestion
        if (connection == null && !locList.isEmpty()) {
```

Kogepan229 commented on src/main/java/appeng/block/networking/BlockWirelessConnector.java:
```diff
@@ -0,0 +1,204 @@
+    @Override
+    public ArrayList<ItemStack> getDrops(World world, int x, int y, int z, int metadata, int fortune) {
+        ItemStack is = new ItemStack(this);
+        TileWirelessConnector te = (TileWirelessConnector) world.getTileEntity(x, y, z);
```
```suggestion
        TileWirelessConnector te = (TileWirelessBase) world.getTileEntity(x, y, z);
```

Kogepan229 commented on src/main/java/appeng/helpers/WireLessToolHelper.java:
```diff
@@ -0,0 +1,278 @@
+            return false;
+        }
+
+        if (target.doLink(otherTile)) {
```
doLink returns true when connecting hubs to each other, but it needs to be handled as a failure.

Kogepan229 commented on src/main/java/appeng/tile/misc/TileCrystalGrowthChamber.java:
```diff
@@ -0,0 +1,272 @@
+
+    @Override
+    public int[] getAccessibleSlotsBySide(ForgeDirection whichSide) {
+        return new int[27];
```
need to return a list of indices for the accessible slots.
like this
```suggestion
        return IntStream.range(0, 27).toArray();
```

Kogepan229 commented on src/main/java/appeng/tile/networking/TileWirelessBase.java:
```diff
@@ -0,0 +1,320 @@
+    }
+
+    public int getUsedChannels() {
+        int used = 0;
```
hubs can have multiple connections.
needs to return the total value.

lc-1337 replied:
<img width="1333" height="1264" alt="image" src="https://github.com/user-attachments/assets/80e7bb85-1439-4a0e-bf62-625597906aaf" />

Kogepan229 commented on src/main/java/appeng/container/implementations/ContainerSuperWirelessKit.java:
```diff
@@ -0,0 +1,594 @@
+        }
+    }
+
+    public void processCommand(SuperWirelessKitCommand command) {
```
need to check BUILD permission

Kogepan229 commented on src/main/java/appeng/helpers/WireLessToolHelper.java:
```diff
@@ -0,0 +1,278 @@
+            return true;
+        }
+
+        if (performConnection(target, tile, p)) {
```
The caller checks the BUILD permission for `target`, but `tile` is not checked.

Kogepan229 commented on src/main/java/appeng/helpers/WireLessToolHelper.java:
```diff
@@ -0,0 +1,278 @@
+                } else break;
+            }
+            p.addChatMessage(WirelessMessages.mode_advanced_binding_hub.toChat(i));
+        } else if (performConnection(target, tile, p)) {
```
The caller checks the BUILD permission for `target`, but `tile` is not checked.

Kogepan229 commented on src/main/java/appeng/container/implementations/ContainerSuperWirelessKit.java:
```diff
@@ -0,0 +1,594 @@
+
+                int i = 0;
+                int ii = 0;
+                while (twToBind.size() > i && twToBind.size() > ii) {
```
maybe?
```suggestion
                while (twToBind.size() > i && twTarget.size() > ii) {
```

Dream-Master committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
rew

lc-1337 committed:
fix convert button tooltips

Ranzuu commented:
super side note, might be already a thing but I cannot find it. Can the blue link line be an enum?

https://github.com/GTNewHorizons/ae2stuff/pull/42#issuecomment-3835294037

lc-1337 committed:
proxy, colors

lc-1337 commented:
@Ranzuu check now

lc-1337 committed:
aebutton colors

Ranzuu commented:
@lc-1337 💋

lc-1337 committed:
visualizer line width config, visualizer texture

lc-1337 committed:
another deps

lc-1337 committed:
text

Dream-Master committed:
Merge branch 'master' into super-wireless-kit

Kogepan229 commented on src/main/java/appeng/client/gui/implementations/GuiSuperWirelessKit.java:
```diff
@@ -0,0 +1,1358 @@
+        public void mouseClicked(final int xPos, final int yPos, final int button) {
+            if (this.isMouseIn(xPos, yPos)) {
+                if (Keyboard.isKeyDown(Keyboard.KEY_LSHIFT)) {
+                    BlockPosHighlighter.highlightBlocks(
```
this `BlockPosHighlighter` is deprecated

Kogepan229 commented on src/main/java/appeng/client/gui/implementations/GuiSuperWirelessKit.java:
```diff
@@ -0,0 +1,1358 @@
+                if (Keyboard.isKeyDown(Keyboard.KEY_LSHIFT)) {
+                    BlockPosHighlighter.highlightBlocks(
+                            mc.thePlayer,
+                            data.targets.get(0) == null ? Collections.singletonList(data.cord)
```
This causes NPE
```suggestion
                            data.targets.isEmpty() ? Collections.singletonList(data.cord)
```

Kogepan229 commented on src/main/java/appeng/client/gui/implementations/GuiSuperWirelessKit.java:
```diff
@@ -0,0 +1,1358 @@
+        }
+    }
+
+    private class baseUnit {
```
```suggestion
    private class BaseUnit {
```

Kogepan229 commented on src/main/java/appeng/client/gui/implementations/GuiSuperWirelessKit.java:
```diff
@@ -0,0 +1,1358 @@
+        }
+    }
+
+    private class groupUnit extends baseUnit {
```
```suggestion
    private class GroupUnit extends baseUnit {
```

Kogepan229 commented on src/main/java/appeng/client/gui/implementations/GuiSuperWirelessKit.java:
```diff
@@ -0,0 +1,1358 @@
+        return subCommand;
+    }
+
+    private static class savedName {
```
```suggestion
    private static class SavedName {
```

Kogepan229 commented on src/main/java/appeng/client/gui/implementations/GuiSuperWirelessKit.java:
```diff
@@ -0,0 +1,1358 @@
+        }
+    }
+
+    private static class savedPin {
```
```suggestion
    private static class SavedPin {
```

Kogepan229 commented:
Incorrect tooltip position
<img width="1920" height="1008" alt="Image" src="https://github.com/user-attachments/assets/52bdc3f4-c38d-4aff-a58f-36c523289c74" />

Kogepan229 commented on src/main/java/appeng/tile/networking/TileWirelessBase.java:
```diff
@@ -0,0 +1,333 @@
+                locList.add(DimensionalCoord.readFromNBT(tag));
+            }
+            tryRestoreConnection(locList);
+            tilesToConnect = null;
```
If the target can’t be connected to, for example because it isn’t loaded, the connection information will be lost.

We need to keep tilesToConnect and retry the connection periodically, like AE2Stuff did.

Also, doing this in the NETWORK_WRITE event seems weird

Kogepan229 commented:
@lc-1337 Do you want to merge this in 2.9?

lc-1337 commented:
@Kogepan229 Would be cool yes

lc-1337 committed:
review

lc-1337 commented:
<img width="1332" height="475" alt="image" src="https://github.com/user-attachments/assets/c3530f22-6bb1-44f6-a649-75158878c36c" />

lc-1337 committed:
fixes, dimensional coord gui colors

lc-1337 committed:
add any network tile in super mode

Kogepan229 commented:
To replace AE2Stuff’s wireless-related features, we also need to port the regular “Wireless Setup Kit”.
Also, I suggest renaming “Super Wireless Kit” to “Advanced Wireless Setup Kit” and using the AE2Stuff texture for it.

This way, we can add only the super mode while leaving the existing behavior mostly unchanged.

lc-1337 commented:
I gonna add simple tools

lc-1337 commented:
I gonna add simple tools, super will stay

Kogepan229 commented:
"Advanced" would fit better for both the name and the texture :(

lc-1337 committed:
add simple and advanced tool
add line mode to advanced tool
rework WirelessMessages

lc-1337 committed:
Merge branch 'master' into super-wireless-kit

lc-1337 commented:
added tools, line mode, some lang stuff and fix reconnection on world load

Kogepan229 commented:
Cool.
What's the reason for adding both the Super kit and the Advanced kit?
I think it would be enough to just add the super mode to the Advanced kit.

Kogepan229 commented:
Even after adding networks, the displayed network list is empty.
<img width="854" height="480" alt="2026-04-29_16 51 31" src="https://github.com/user-attachments/assets/b88be568-f39f-4bea-9fde-dfd5142cb7c1" />

Chat messages are not localized.
<img width="854" height="480" alt="2026-04-29_16 51 35" src="https://github.com/user-attachments/assets/92714c5c-a75e-4635-a972-0be82814b074" />

Kogepan229 commented on src/main/java/appeng/items/tools/ToolSuperWirelessKit.java:
```diff
@@ -0,0 +1,239 @@
+
+    public static IConfigManager getConfigManager(final ItemStack target) {
+        final ConfigManager out = new ConfigManager((manager, settingName, newValue) -> {
+            final NBTTagCompound data = Platform.openNbtData(target);
```
Since openNbtData is deprecated now, use the methods on ItemStackNBT instead

Kogepan229 commented:
Recoloring groups with the wireless kit in super mode causes a crash.

Kogepan229 commented on src/main/resources/assets/appliedenergistics2/lang/en_US.lang:
```diff
@@ -738,6 +753,34 @@ gui.tooltips.appliedenergistics2.CPUOnlyAllowNonPlayerDesc=Only allow crafting r
+gui.tooltips.appliedenergistics2.SuperWirelessKitIncludeConnectorsName=Include hubs
+gui.tooltips.appliedenergistics2.SuperWirelessKitIncludeConnectorsDesc=Include hubs in manipulations
+gui.tooltips.appliedenergistics2.SuperWirelessKitIncludeHubsName=Include connectors
+gui.tooltips.appliedenergistics2.SuperWirelessKitIncludeHubsDesc=Include connectors in manipulations
```
Name and content are reversed.
But it is displayed correctly in the GUI.

Kogepan229 commented on src/main/java/appeng/helpers/WireLessToolHelper.java:
```diff
@@ -0,0 +1,518 @@
+
+    private static boolean bindFromQueue(TileWirelessBase target, ItemStack tool, World w, EntityPlayer p) {
+        List<DimensionalCoord> locList = DimensionalCoord
+                .readAsListFromNBT(tool.getTagCompound().getCompoundTag("advanced"));
```
```suggestion
                .readAsListFromNBT(tool.getTagCompound().getCompoundTag(NbtAdvanced));
```

Kogepan229 commented on src/main/java/appeng/helpers/WireLessToolHelper.java:
```diff
@@ -0,0 +1,518 @@
+
+        final NBTTagCompound tag = new NBTTagCompound();
+        DimensionalCoord.writeListToNBT(tag, locList);
+        tool.getTagCompound().setTag("advanced", tag);
```
```suggestion
        tool.getTagCompound().setTag(NbtAdvanced, tag);
```

Kogepan229 commented on src/main/java/appeng/helpers/WireLessToolHelper.java:
```diff
@@ -0,0 +1,518 @@
+
+    private static boolean addToQueue(TileWirelessBase target, ItemStack tool, EntityPlayer p) {
+        List<DimensionalCoord> locList = DimensionalCoord
+                .readAsListFromNBT(tool.getTagCompound().getCompoundTag("advanced"));
```
```suggestion
                .readAsListFromNBT(tool.getTagCompound().getCompoundTag(NbtAdvanced));
```

Kogepan229 commented on src/main/java/appeng/helpers/WireLessToolHelper.java:
```diff
@@ -0,0 +1,518 @@
+
+        NBTTagCompound tag = new NBTTagCompound();
+        DimensionalCoord.writeListToNBT(tag, locList);
+        tool.getTagCompound().setTag("advanced", tag);
```
```suggestion
        tool.getTagCompound().setTag(NbtAdvanced, tag);
```

lc-1337 commented:
> Cool.
> What's the reason for adding both the Super kit and the Advanced kit?
> I think it would be enough to just add the super mode to the Advanced kit.

At first idea was add a single tool with all functional. But if we add some tool functinal progression, then it would looks better if we get one functinal type per level:
Simple - connector to connector
Advanced - memory, mb wroth add simple mode
Super - all before and many other

Kogepan229 commented:
I don’t think it makes sense to have such fine-grained tiering for what’s basically just a QoL tool.
Maybe a single tool, like in your first idea, would be enough.

lc-1337 committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
fixes

lc-1337 commented:
fixed

lc-1337 committed:
lighter color for DimensionCoord gui text
local var instead of repeat formula

lc-1337 committed:
fix newnbt

Dream-Master committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
postea

lc-1337 committed:
fix color

lc-1337 committed:
Merge branch 'master' into super-wireless-kit

Dream-Master committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
tools tooltips

Kogepan229 commented:
As discussed before, remove Simple and Advanced.
Rename the "Super Wireless Kit" to "Wireless Kit", and use the same texture as the original.
We only need one tool, and the name of the texture should not be changed from the original.

lc-1337 committed:
sad spiderman walks

lc-1337 commented:
done

Kogepan229 commented on src/main/java/appeng/api/util/DimensionalCoord.java:
```diff
@@ -113,6 +116,37 @@ public static List<DimensionalCoord> readAsListFromNBT(final NBTTagCompound tag)
+        data.writeInt(this.dimId);
+    }
+
+    public static void writeListToPacket(final ByteBuf data, final List<NamedDimensionalCoord> list) {
```
Method for NamedDimensionalCoord should not be here.

Kogepan229 commented on src/main/java/appeng/api/config/Settings.java:
```diff
@@ -108,6 +124,23 @@ public enum Settings {
         this.values = possibleOptions;
     }
 
+    public static Settings getFromClass(Class<? extends Enum<?>> clazz) {
```
Settings do not have a one-to-one relationship with enum classes.
For example, YesNo is used by many settings, such as SEARCH_TOOLTIPS, BLOCK, SMART_BLOCK, and SUPER_WIRELESS_TOOL_HIDE_BOUNDED.

I would recommend either passing both the enum class and the Setting as arguments, or keeping the current approach of retrieving it from ConfigManager and manually casting it afterward.

Kogepan229 commented on src/main/java/appeng/api/config/WirelessToolType.java:
```diff
@@ -0,0 +1,20 @@
+import appeng.core.localization.Localization;
+import appeng.core.localization.WirelessMessages;
+
+public enum WirelessToolType implements Localization {
```
WirelessToolMode would be a better name

Kogepan229 reviewed (commented):
Could you also remove “Super” from the class and method names?
Once these changes are done, we can merge this PR.

lc-1337 committed:
.

lc-1337 commented:
done

Kogepan229 commented:
LGTM!

Kogepan229 commented on src/main/java/appeng/core/localization/GuiColors.java:
```diff
@@ -218,7 +219,29 @@ public enum GuiColors implements Localization {
+    WirelessKitBad(AEColor.Red.mediumVariant),
+
+    NetworkVisualiserFloatingText(0xffffffff),
+    NetworkVisualiserNodeMissing(0xffff0000),
+    NetworkVisualiserNodeDense(0xffffff00),
+    NetworkVisualiserNodeProxy(0xffffa500),
+    NetworkVisualiserNodeDefault(0xff0000ff),
+
+    NetworkVisualiserLinkCompressed(0xffff00ff),
+    NetworkVisualiserLinkDense(0xffffff00),
+    NetworkVisualiserLinkProxy(0xffffa500),
+    NetworkVisualiserLinkDefault(0xff0000ff),
```
ah these aren't for GUI, so it might be better to create a separate enum.

lc-1337 replied:
for 8 lines?

Kogepan229 replied:
It may increase in the future.
But not a blocker, feel free to ignore it.

PLASMAchicken committed:
CI opti images

Dream-Master committed:
Merge branch 'master' into super-wireless-kit

Kogepan229 committed:
Port Advanced Inscriber

Kogepan229 committed:
Fix postea id

Kogepan229 committed:
Fix wireless kit texture name

Kogepan229 committed:
Make connection handling closer to ae2stuff and more robust

Kogepan229 committed:
Ingnore missing mappings

Kogepan229 committed:
Fix crystal growth chamber

Match Crystal Growth Chamber refresh behavior to ae2stuff by preventing tile refresh on metadata-only changes.
Change Crystal Growth Chamber cycle power from 10 AE to 100 AE per speed factor.
Limit Fluix crystal creation to once per processing cycle.
Insert Fluix output through AE2 inventory handling instead of manually increasing stack size.
Require full Fluix output insertion before consuming ingredients.
Wake Crystal Growth Chamber ticking when the grid changes after world load or network rebuild.

lc-1337 committed:
Merge branch 'master' into super-wireless-kit

Dream-Master committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
local color name on color button

Dream-Master committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
spotless

Dream-Master committed:
Merge branch 'master' into super-wireless-kit

lc-1337 committed:
move wireless to core

Dream-Master committed:
Merge branch 'master' into super-wireless-kit
